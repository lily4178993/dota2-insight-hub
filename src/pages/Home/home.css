.home-container {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}

.home-container a {
  width: 25%;
  color: var(--color-white);
}

@media (max-width: 768px) {
  .home-container a {
    width: 50%;
  }
}
