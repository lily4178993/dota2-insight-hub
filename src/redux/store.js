import { configureStore/* , getDefaultMiddleware */ } from '@reduxjs/toolkit';
/* import logger from 'redux-logger'; */
import heroReducer from './slices/heroesSlice';
import itemReducer from './slices/itemsSlice';
import matchReducer from './slices/matchesSlice';
import playerReducer from './slices/playersSlice';

const store = configureStore({
  reducer: {
    heroes: heroReducer,
    items: itemReducer,
    matches: matchReducer,
    players: playerReducer,
  },
  /* middleware: [...getDefaultMiddleware(), logger], */
});

export default store;
