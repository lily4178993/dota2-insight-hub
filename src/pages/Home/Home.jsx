import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import {
  fetchHeroes,
  fetchItems,
  fetchMatches,
  fetchPlayers,
  selectHeroesState,
  selectItemsState,
  selectMatchesState,
  selectPlayersState,
} from '../../redux/slices';
import { Card } from '../../components';
import urlSpaceChecker from '../../utils/urlSpaceChecker';
import heroesImage from '../../assets/angelSword.png';
import itemsImage from '../../assets/hammer.png';
import matchesImage from '../../assets/trophy.png';
import playersImage from '../../assets/gamepad.png';
import './home.css';

function Home() {
  const { heroes, loading: loadingHeroes, error: errorHeroes } = useSelector(selectHeroesState);
  const { items, loading: loadingItems, error: errorItems } = useSelector(selectItemsState);
  const { matches, loading: loadingMatches, error: errorMatches } = useSelector(selectMatchesState);
  const { players, loading: loadingPlayers, error: errorPlayers } = useSelector(selectPlayersState);
  const dispatch = useDispatch();

  useEffect(() => {
    if (loadingHeroes === false && heroes.length === 0) {
      dispatch(fetchHeroes());
    }
    if (loadingItems === false && items.length === 0) {
      dispatch(fetchItems());
    }
    if (loadingMatches === false && matches.length === 0) {
      dispatch(fetchMatches());
    }
    if (loadingPlayers === false && players.length === 0) {
      dispatch(fetchPlayers());
    }
  }, [
    dispatch,
    loadingHeroes,
    loadingItems,
    loadingPlayers,
    players.length,
    loadingMatches,
    matches.length,
    heroes.length,
    items.length,
  ]);

  const data = [
    {
      cardTitle: 'Heroes',
      cardImage: heroesImage,
      cardCount: loadingHeroes ? 'Loading...' : heroes.length,
    },
    {
      cardTitle: 'Items',
      cardImage: itemsImage,
      cardCount: loadingItems ? 'Loading...' : items.length,
    },
    {
      cardTitle: 'Pro Matches',
      cardImage: matchesImage,
      cardCount: loadingMatches ? 'Loading...' : matches.length,
    },
    {
      cardTitle: 'Pro Players',
      cardImage: playersImage,
      cardCount: loadingPlayers ? 'Loading...' : players.length,
    },
  ];
  return (
    <section>
      <div className="home-container">
        {data.map((cardData) => (
          <Link to={`/details/${urlSpaceChecker(`${cardData.cardTitle}`)}`} key={cardData.cardTitle}>
            <Card
              cardImage={cardData.cardImage}
              cardTitle={cardData.cardTitle}
              cardCount={cardData.cardCount}
            />
          </Link>
        ))}
      </div>
      {errorHeroes && <p>{errorHeroes}</p>}
      {errorItems && <p>{errorItems}</p>}
      {errorMatches && <p>{errorMatches}</p>}
      {errorPlayers && <p>{errorPlayers}</p>}
    </section>
  );
}

export default Home;
