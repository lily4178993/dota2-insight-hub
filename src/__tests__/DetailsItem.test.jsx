import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../pages/Home/Home';
import store from '../redux/store';

test('Home component snapshot', () => {
  const { asFragment } = render(
    <Router>
      <Provider store={store}>
        <Home />
      </Provider>
    </Router>,
  );

  expect(asFragment()).toMatchSnapshot();
});
