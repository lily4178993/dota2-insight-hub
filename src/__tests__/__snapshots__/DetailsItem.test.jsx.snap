// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component snapshot 1`] = `
<DocumentFragment>
  <section>
    <div
      class="header-filter-box"
    >
      <span>
        Choose a Card
      </span>
      <button
        title="Filter"
        type="button"
      >
        Filter 
        <i
          class="bx bx-down-arrow-circle"
          style="transform: rotate(0deg);"
        />
      </button>
      <div
        class="filter-box"
        style="right: -100%;"
      >
        <ul>
          <li>
            <button
              title="Filter by Names"
              type="button"
            >
              Names
            </button>
          </li>
          <li>
            <button
              title="Filter by Counts"
              type="button"
            >
              Counts
            </button>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="home-container"
    >
      <a
        href="/details/heroes"
      >
        <div
          class="card-container"
        >
          <button
            title="More details"
            type="button"
          >
            <i
              class="bx bx-right-arrow-circle"
            />
          </button>
          <div
            class="card-image"
          >
            <img
              alt="Heroes"
              src="angelSword.png"
            />
          </div>
          <div
            class="card-content"
          >
            <h2>
              Heroes
            </h2>
            <span>
              0
            </span>
          </div>
        </div>
      </a>
      <a
        href="/details/items"
      >
        <div
          class="card-container"
        >
          <button
            title="More details"
            type="button"
          >
            <i
              class="bx bx-right-arrow-circle"
            />
          </button>
          <div
            class="card-image"
          >
            <img
              alt="Items"
              src="hammer.png"
            />
          </div>
          <div
            class="card-content"
          >
            <h2>
              Items
            </h2>
            <span>
              0
            </span>
          </div>
        </div>
      </a>
      <a
        href="/details/proMatches"
      >
        <div
          class="card-container"
        >
          <button
            title="More details"
            type="button"
          >
            <i
              class="bx bx-right-arrow-circle"
            />
          </button>
          <div
            class="card-image"
          >
            <img
              alt="Pro Matches"
              src="trophy.png"
            />
          </div>
          <div
            class="card-content"
          >
            <h2>
              Pro Matches
            </h2>
            <span>
              0
            </span>
          </div>
        </div>
      </a>
      <a
        href="/details/proPlayers"
      >
        <div
          class="card-container"
        >
          <button
            title="More details"
            type="button"
          >
            <i
              class="bx bx-right-arrow-circle"
            />
          </button>
          <div
            class="card-image"
          >
            <img
              alt="Pro Players"
              src="gamepad.png"
            />
          </div>
          <div
            class="card-content"
          >
            <h2>
              Pro Players
            </h2>
            <span>
              0
            </span>
          </div>
        </div>
      </a>
    </div>
  </section>
</DocumentFragment>
`;
